src/app/api/routes/devices.py

```diff
@@ -3,17 +3,20 @@
 from fastapi import APIRouter, Path, Security, HTTPException
 
 from app.api import crud
-from app.db import devices
+from app.db import database, devices
 from app.api.schemas import DeviceOut, DeviceAuth, DeviceCreation, DeviceIn, UserRead, DefaultPosition, Cred
 from app.api.deps import get_current_device, get_current_user
 
-from app.api.routes.accesses import post_access, update_access_pwd
+from app.api.crud.accesses import post_access, update_access_pwd
+
 
 router = APIRouter()
 
 
 @router.post("/", response_model=DeviceOut, status_code=201)
+@database.transaction()
```

Comment:
Why add this decorator only for route that create objects?

Reply:
I planned to add the decorator when transaction is needed (writing in several tables with atomicity), user/device creation was the first use case, did I miss other ones ?

I realised that route decorator was not the best practice: it makes it impossible to test. We may need to find a way to mock a database transaction or we'll have to rewrite operations into pytest mocks. @frgfm do you see what I mean ?

Reply:
I am no where close to be knowledgeable about database transactions since I just discovered that thanks to you @jeanpasquier75. What is the expected behaviour difference after decorating the route? reducing execution time? output type change?

If it's beneficial to the API, even if we can't test it per se, I would argue that we should still put it. If it doesn't bring much and we can't test it, then perhaps let's leave it out